### PR TITLE
Addition of Mass Storage Driver to this library

### PR DESCRIPTION
**Pull request by wwatson4506**
Hi Paul,

I finally got the nerve to submit the latest version of the MSC driver for inclusion into USBHost_t36.
All I added was the "MassStorageDriver.cpp" and modified "msc.h" files plus one test sketch.
"MassStorageDriver.cpp" was added to the main directory, "msc.h" now contains only the needed parts of "MassStorage.h" (not included). The contents of "msc.h" were added to "USBHost_t36.h". I put "msc.h" file in the utility directory. I am not sure if this is the proper place to put it. the only ".h" files I saw in the main directory were def.h files.
An example sketch, "mscTesting.ino" was added to the examples directory.

Thanks

### CODE REVIEW

**Comment by FrankBoesing** [comment]
wow.

**Comment by PaulStoffregen** [comment]
I'd like to consider some changes to the public API before we have a non-beta release.  But merging this now, we everyone can start using it.

**Comment by wwatson4506** [comment]
Sounds great. I am open to to any suggestions or ideas you have in mind. I will follow your lead on this as I do not have the level of programming or software engineering skills as you and most of the top level programmers on the forum do.

I am all ears:)

File: MassStorageDriver.cpp [hunk 1]
```diff
@@ -0,0 +1,596 @@
...
+	return msResult;
+}
+
+//---------------------------------------------------------------------------
+// Perform Mass Storage Reset
+void msController::msReset() {
+#ifdef DBGprint
+	Serial.printf("msReset()\n");
+#endif
+	mk_setup(setup, 0x21, 0xff, 0, 0, 0);
```
Comment (hathach) [inline MassStorageDriver.cpp:235]: Hi, I am late to the party, just want to say the `wIndex` should  be the interface number. This currently only work with MSC as the first interface (itf num = 0). Same applies to `msGetMaxLun()`. I am not familiar with codebase to submit the PR.

**Pull request merged**